File: web-front-reactive/src/main/java/template/modules/admin/query/users/UserFetcher.java
package template.modules.admin.query.users;

import lombok.AllArgsConstructor;
import org.springframework.stereotype.Service;
import reactor.core.publisher.Flux;
import reactor.core.publisher.Mono;
import template.modules.admin.domain.users.User;
import template.modules.admin.domain.users.UserRepository;

@Service
@AllArgsConstructor
public class UserFetcher {
    private UserRepository userRepository;

    public Flux<User> findAll() {
        return userRepository.findAll();
    }

    public Mono<User> findById(Long id) {
        return userRepository.findById(id);
    }

    public Mono<User> findByEmail(String email) {
        return userRepository.findByEmail(email).elementAt(0);
    }

}

File: web-front-reactive2/src/main/kotlin/template/modules/admin/query/users/Userfetcher.kt
package template.modules.admin.query.users

import kotlinx.coroutines.reactive.asFlow
import kotlinx.coroutines.reactive.awaitFirstOrNull
import org.springframework.stereotype.Service
import template.modules.admin.domain.users.UserRepository

@Service
class UserFetcher(private val repo: UserRepository) {
    suspend fun findAll() = repo.findAll().asFlow()
    suspend fun findById(id: Long) = repo.findById(id).awaitFirstOrNull()
    suspend fun findByEmail(email: String) = repo.findByEmail(email).asFlow()
}